pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/buffer/BufferedDataTestBase.java

```diff
@@ -106,6 +106,22 @@ void sliceThenReadVarInt(final int num) {
         assertEquals(afterSecondIntPos - afterFirstIntPos, slicedBuf.position());
     }
 
+    @ParameterizedTest
+    @ValueSource(ints = {0, 1, 2, 4, 7, 8, 15, 16, 31, 32, 33, 127, 128, 512, 1000, 1024, 4000,
+            16384, 65535, 65536, 65537, 0xFFFFFF, 0x1000000, 0x1000001, 0x7FFFFFFF,
+            -1, -7, -8, -9, -127, -128, -129, -65535, -65536, -0xFFFFFF, -0x1000000, -0x1000001, -0x80000000})
+    @DisplayName("readVar() won't read beyond 10 bytes")
+    void readVarFromLargeBuffer(final int num) {
+        final var buf = allocate(100);
```

Comment:
Var ints (including longs) may not take more than 10 bytes

Reply:
Yes, yes, but as the name of the test suggests it specifically wants to use a longer buffer to check an edge case somewhere in the implementation. 100 is long enough for me.

That edge case may have existed in the old readVarInt implementation and perhaps you removed it when you overhauled its implementation a few weeks ago. However, the test seems good still - it still covers the edge case and/or will cover it if/when we re-introduce it in the future.